test/select.spec.js

```diff
@@ -636,6 +639,31 @@ describe('ui-select tests', function () {
     expect(el.find('.select2-search-choice-close').length).toEqual(1);
   });
 
+  it('should clear selection (with object as source)', function() {
```

Comment:
ah missed that yeah hmm.. maybe we could make a new function `createUiSelectObjectSource` or something where the template is build and can be used in the future?